time-benchmarks/benches/rand.rs

```diff
@@ -0,0 +1,27 @@
+use bench_util::setup_benchmark;
+use criterion::BatchSize;
+use rand::Rng;
+use time::{Date, Duration, OffsetDateTime, PrimitiveDateTime, Time, UtcOffset, Weekday};
+
+setup_benchmark! {
+    "Random",
+
+    rng(ben: &mut Bencher) {
+        ben.iter_batched_ref(
+            // Use 1<<29 to work around rust-random/rand#1020.
+            || rand::rngs::mock::StepRng::new(0, 1 << 29),
+            |rng| {
+                let _ = rng.gen::<Time>();
+                let _ = rng.gen::<Date>();
+                let _ = rng.gen::<UtcOffset>();
+                let _ = rng.gen::<PrimitiveDateTime>();
+                let _ = rng.gen::<OffsetDateTime>();
+                let _ = rng.gen::<Duration>();
+                for _ in 0..7 {
+                    let _ = rng.gen::<Weekday>();
+                }
```

Comment:
This `for` loop is only present for test coverage, and can be removed.

```suggestion
                let _ = rng.gen::<Weekday>();
```